src/profile-logic/marker-data.js

```diff
@@ -1375,26 +1357,38 @@ export function getMarkerTypesForDisplay(
   return types;
 }
 
+function _doNotAutomaticallyAdd(_data: Marker) {
+  return undefined;
+}
+
 /**
- * Filter markers to a smaller set based on what "type" they are. This is used to
- * to filter markers based on the schema.
+ * Filter markers to a smaller set based on the location.
  */
-export function filterMarkerByTypes(
+export function filterMarkerByDisplayLocation(
   getMarker: MarkerIndex => Marker,
   markerIndexes: MarkerIndex[],
-  markerTypes: Set<string>
+  markerSchema: MarkerSchema[],
+  displayLocation: MarkerDisplayLocation,
+  // This argument allows a filtering function to customize the result, without having
+  // to loop through all of the markers again. Return a boolean if making a decision,
+  // or undefined if not.
+  preemptiveFilterFunc?: (
+    data: Marker
+  ) => boolean | void = _doNotAutomaticallyAdd
 ): MarkerIndex[] {
+  const markerTypes = getMarkerTypesForDisplay(markerSchema, displayLocation);
   return filterMarkerIndexes(getMarker, markerIndexes, marker => {
-    const { data } = marker;
-    if (!data) {
-      return false;
-    }
-    if (data.type === 'tracing') {
-      // This is a bit gross, but tracing markers are all of type "tracing", but
-      // differentiate on their categories. This should probably be fixed, but
-      // for now just do another check here.
-      return markerTypes.has(data.category);
+    const additionalResult = preemptiveFilterFunc(marker);
+
+    if (additionalResult !== undefined) {
+      // This is a boolean value, use it rather than the schema.
+      return additionalResult;
     }
-    return markerTypes.has(data.type);
+
+    return markerTypes.has(getMarkerSchemaName(marker));
   });
 }
+
+export function filterOverviewMarkers(_marker: Marker): boolean | void {
+  return undefined;
+}
```

Comment:
Is it a leftover? This doesn't seem useful :-)

Reply:
Haha.

Reply:
Actually, it is needed by the selector, I could inline it, but I think it's clearer to keep it. I'll add more supplementary comments.

Reply:
Well I keep changing my mind on myself. That last parameter is optional, so I can remove it.